sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/util/ModelHelper.java

```diff
@@ -241,4 +302,311 @@ public static String decodeName(StringEncoded stringEncoded) {
+     */
+    public static void validateFilePermissionAndKey(String filePermission, String  filePermissionKey) {
+        if (filePermission != null && filePermissionKey != null) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                FileConstants.MessageConstants.FILE_PERMISSION_FILE_PERMISSION_KEY_INVALID));
+        }
+
+        if (filePermission != null) {
+            StorageImplUtils.assertInBounds("filePermission",
+                filePermission.getBytes(StandardCharsets.UTF_8).length, 0, 8 * Constants.KB);
+        }
+    }
+
+    public static boolean checkDoesNotExistStatusCode(Throwable t) {
+        // ShareStorageException
+        return (t instanceof ShareStorageException
+            && ((ShareStorageException) t).getStatusCode() == 404
+            && (((ShareStorageException) t).getErrorCode() == ShareErrorCode.RESOURCE_NOT_FOUND
+            || ((ShareStorageException) t).getErrorCode() == ShareErrorCode.SHARE_NOT_FOUND))
+
+            /* HttpResponseException - file get properties is a head request so a body is not returned. Error
+             conversion logic does not properly handle errors that don't return XML. */
+            || (t instanceof HttpResponseException
+            && ((HttpResponseException) t).getResponse().getStatusCode() == 404
+            && (((HttpResponseException) t).getResponse().getHeaderValue("x-ms-error-code")
+            .equals(ShareErrorCode.RESOURCE_NOT_FOUND.toString())
+            || (((HttpResponseException) t).getResponse().getHeaderValue("x-ms-error-code")
+            .equals(ShareErrorCode.SHARE_NOT_FOUND.toString()))));
+    }
+
+    public static Response<ShareFileInfo> createFileInfoResponse(ResponseBase<FilesCreateHeaders, Void> response) {
+        String eTag = response.getDeserializedHeaders().getETag();
+        OffsetDateTime lastModified = response.getDeserializedHeaders().getLastModified();
+        boolean isServerEncrypted = response.getDeserializedHeaders().isXMsRequestServerEncrypted();
+        FileSmbProperties smbProperties = new FileSmbProperties(response.getHeaders());
+        ShareFileInfo shareFileInfo = new ShareFileInfo(eTag, lastModified, isServerEncrypted, smbProperties);
+        return new SimpleResponse<>(response, shareFileInfo);
+    }
+
+    public static Response<ShareFileProperties> getPropertiesResponse(
+        final ResponseBase<FilesGetPropertiesHeaders, Void> response) {
+        FilesGetPropertiesHeaders headers = response.getDeserializedHeaders();
+        String eTag = headers.getETag();
+        OffsetDateTime lastModified = headers.getLastModified();
+        Map<String, String> metadata = headers.getXMsMeta();
+        String fileType = headers.getXMsType();
+        Long contentLength = headers.getContentLength();
+        String contentType = headers.getContentType();
+        byte[] contentMD5;
+        try {
+            contentMD5 = headers.getContentMD5();
```

Comment:
Why would this produce a NullPointerException? That seems like a bug in the getter.